src/tools/mem_use_watcher/scripts/watcher.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+import rospy
+from std_msgs.msg import Bool
+import psutil
+
+
+def monitorAvailableMemory(memory_upperlimit_percent):
+    """
+    This function is used to monitor the memory utilization and throw an error 
+    if it exceeds a preset value.
+
+    Arguments:
+    memory_upperlimit_percent: The upperlimit of the memory utilization (float)
+    """
+
+    # Utilized memory
+    utilized_memory = psutil.virtual_memory().percent
+
+    if utilized_memory > memory_upperlimit_percent:
+        return True
+
+    return False
+
+
+def mem_use_watcher():
```

Comment:
It would be good to change the name to make it specific to RAM. The same goes for the topic.

Reply:
I used memory in place of RAM since memory is a synonym for RAM. Do you want me to change that to RAM instead of memory?

Reply:
Either is fine.